From 38f1a9995638ec50407eab74b52b6c8df308cb6c Mon Sep 17 00:00:00 2001
From: Dylan Yip <dylan.yip@xilinx.com>
Date: Wed, 30 Sep 2020 13:19:06 -0700
Subject: [PATCH 1683/1851] media: videodev2.h: Add control type for HDR

commit 67b73df960281b2bc17aa88e0af10fb33b98323c from
https://github.com/Xilinx/linux-xlnx.git

V4L2 core will overwrite certain control variables depending on
v4l2_ctrl_type. Previously we hardcoded v4l2_ctrl_type but this was
giving coverity issues: 'mixed_enum_type'. To avoid this, add HDR
compound type.

There is a large gap between V4L2_CTRL_TYPE_U32 and V4L2_CTRL_TYPE_HDR
to avoid any future enum conflicts because HDR controls are not
upstreamed yet.

While at it, also add V4L2_CTRL_TYPE_HDR to documentation.

Signed-off-by: Dylan Yip <dylan.yip@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 Documentation/media/uapi/v4l/vidioc-queryctrl.rst | 6 ++++++
 include/uapi/linux/videodev2.h                    | 1 +
 2 files changed, 7 insertions(+)

diff --git a/Documentation/media/uapi/v4l/vidioc-queryctrl.rst b/Documentation/media/uapi/v4l/vidioc-queryctrl.rst
index dc500632095d..6d9a9878dd22 100644
--- a/Documentation/media/uapi/v4l/vidioc-queryctrl.rst
+++ b/Documentation/media/uapi/v4l/vidioc-queryctrl.rst
@@ -473,6 +473,12 @@ See also the examples in :ref:`control`.
       - n/a
       - A struct :c:type:`v4l2_ctrl_h264_decode_params`, containing H264
 	decode parameters for stateless video decoders.
+    * - ``V4L2_CTRL_TYPE_HDR``
+      - n/a
+      - n/a
+      - n/a
+      - A struct :c:type:`v4l2_metadata_hdr`, containing HDR
+	metadata.
 
 .. tabularcolumns:: |p{6.6cm}|p{2.2cm}|p{8.7cm}|
 
diff --git a/include/uapi/linux/videodev2.h b/include/uapi/linux/videodev2.h
index dadfe7b642bc..12fe824fa09c 100644
--- a/include/uapi/linux/videodev2.h
+++ b/include/uapi/linux/videodev2.h
@@ -1744,6 +1744,7 @@ enum v4l2_ctrl_type {
 	V4L2_CTRL_TYPE_U8	     = 0x0100,
 	V4L2_CTRL_TYPE_U16	     = 0x0101,
 	V4L2_CTRL_TYPE_U32	     = 0x0102,
+	V4L2_CTRL_TYPE_HDR	     = 0x0200,
 };
 
 /*  Used in the VIDIOC_QUERYCTRL ioctl for querying controls */
-- 
2.31.1

